### PR TITLE
Fix to automatic deriving of units when saving in netcdf format

### PR DESCRIPTION
**Pull request by djkirkham**
Make coords with standard name 'latitude' and 'longitude' which have no coordsystem to be saved with units 'degree_north' and 'degree_east' respectively in netcdf files. Remove other unsafe unit changes.

Change some unit test expected output to recognise these changes.

### CODE REVIEW

**Comment by marqh** [comment]
please add a what's new contribution for this change

File: lib/iris/fileformats/netcdf.py [hunk 1]
```diff
@@ -1119,20 +1119,13 @@ def _cf_coord_identity(self, coord):
 
         units = str(coord.units)
 
-        # TODO: Use #61 to get the units.
-        if isinstance(coord.coord_system, iris.coord_systems.GeogCS):
-            if "latitude" in coord.standard_name:
+        if ((isinstance(coord.coord_system, iris.coord_systems.GeogCS) or
```
Comment (marqh) [inline lib/iris/fileformats/netcdf.py]: perhaps an in-line comment here would help a little.  I would suggest

Set the 'units' of 'latitude' and 'longitude' coordinates specified in 'degrees' to 'degrees_north' and 'degrees_east' respectively, as defined in the CF conventions for netCDF files: sections 4.1 and 4.2.

File: lib/iris/tests/results/netcdf/netcdf_save_ndim_auxiliary.cdl [hunk 2]
```diff
@@ -36,11 +36,11 @@ variables:
 		rlon:standard_name = "grid_longitude" ;
 		rlon:long_name = "rotated longitude" ;
 	float lat(rlat, rlon) ;
-		lat:units = "degrees" ;
+		lat:units = "degrees_north" ;
```
Comment (marqh) [inline lib/iris/tests/results/netcdf/netcdf_save_ndim_auxiliary.cdl]: this is the test result I want to see change, good

File: lib/iris/tests/results/netcdf/netcdf_save_wcoord.cdl [hunk 3]
```diff
@@ -8,7 +8,7 @@ variables:
 		temp:units = "K" ;
 	int64 lon(lon) ;
 		lon:axis = "X" ;
-		lon:units = "degree_east" ;
+		lon:units = "degrees_east" ;
```
Comment (marqh) [inline lib/iris/tests/results/netcdf/netcdf_save_wcoord.cdl]: that's an oddity, but the result is better

File: lib/iris/tests/results/netcdf/netcdf_save_gridmapmulti.cdl [hunk 4]
```diff
@@ -15,12 +15,12 @@ variables:
 		latitude_longitude:earth_radius = 6371229. ;
 	int64 longitude(longitude) ;
 		longitude:axis = "X" ;
-		longitude:units = "degrees_east" ;
+		longitude:units = "1" ;
```
Comment (marqh) [inline lib/iris/tests/results/netcdf/netcdf_save_gridmapmulti.cdl]: i'm surprised by the test result changes in this file.  I would have expected all of these to stay the same

@djkirkham why do you think this has occurred?

Comment (djkirkham) [inline lib/iris/tests/results/netcdf/netcdf_save_gridmapmulti.cdl]: It appears the units of these coords are '1', so they fail the if check. I'll look into why they have units of '1'.

Comment (djkirkham) [inline lib/iris/tests/results/netcdf/netcdf_save_gridmapmulti.cdl]: In the test these coordinates are created from numpy arrays without specifying a unit, so it defaults to '1'.

Comment (marqh) [inline lib/iris/tests/results/netcdf/netcdf_save_gridmapmulti.cdl]: which test is it?
may I suggest that perhaps a more useful test would be where they are created with a units of 'degrees'?

Comment (djkirkham) [inline lib/iris/tests/results/netcdf/netcdf_save_gridmapmulti.cdl]: It's this test: https://github.com/SciTools/iris/blob/30f3e2dc62308f1abdffa055e177ad746ef6d075/lib/iris/tests/test_netcdf.py#L599

> may I suggest that perhaps a more useful test would be where they are created with a units of 'degrees'?

I agree

**Comment by marqh** [comment]
looking good @djkirkham I think this is ready to be adopted

please may you squish the commits?

**Comment by marqh** [comment]
excellent, thanks @djkirkham

**Pull request merged**